src/dsk/alloc.c

```diff
@@ -0,0 +1,34 @@
+#include "alloc.h"
+#include "mdisk.h"
+#include "../mkfs/metaBlocks.h"
+
+static pthread_mutex_t iNodeListMutex = PTHREAD_MUTEX_INITIALIZER;
+const int SLEEP_TIME_IN_SECONDS = 3;
+
+size_t alloc(superBlock* superBlock) {
```

Comment:
`alloc` would be called by a higher level module, which will have to fetch the superblock if `alloc` expects it. Let `alloc` fetch the superblock itself, instead of passing it in.